### PR TITLE
Refactor tests to work with testcontainers

### PR DESCRIPTION
**Pull request by flimzy**

### CODE REVIEW

**Comment by flimzy** [comment]
@Yiling-J Have a look.  This is now working, though I have some additional cleanups to do before it's ready to merge.  But I'm happy to get your preliminary thoughts.

No longer is the use of docker-compose needed for testing.  Instead, for the simple case, simply run:

```bash
MYSQL_DSN=auto PSQL_DSN=auto go test ./...
```

If you want to use an existing, persistent, database (i.e. to avoid the startup time of a docker container for rapid testing), you can do that by providing the dsn:

```bash
MYSQL_DSN=root:pass@/ PSQL_DSN=postgres://postgres:pass@localhost/ go test ./...
```

Or if you want to test only against MySQL for example:

```bash
MYSQL_DSN=auto go test ./...
```

I've also updated CI to no longer use services in GitHub actions. While there's nothing wrong with that approach, I like excercising the testcontainers approach in CI as well, just to keep local and CI as similar as possible.

Curious to hear your reaction.

File: bootstrap_test.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+	}
+	if _, err := db.Exec("CREATE DATABASE " + testDB); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func startPostgres(t *testing.T) string {
+	ctx := context.Background()
+
+	postgresContainer, err := postgres.RunContainer(ctx,
```
Comment (Yiling-J) [inline bootstrap_test.go:75]: just take a look testcontainer's doc and seems you need to terminate container after test? Maybe with the help of [test.Main](https://pkg.go.dev/testing#hdr-Main)

Comment (flimzy) [reply bootstrap_test.go:75]: It will automatically be cleaned up after a few seconds.

**Comment by flimzy** [comment]
I just went through my PR agian, to look for any cleanups, and didn't find much more than some typo corrections and clarifications. I think this is ready for a proper review now.

**Comment by Yiling-J** [comment]
> I just went through my PR agian, to look for any cleanups, and didn't find much more than some typo corrections and clarifications. I think this is ready for a proper review now.

lint failed because of golangci-lint version? I already merge that PR

**Pull request merged**